src/clj/tic_tac_toe_clojure/core.clj

```diff
@@ -57,14 +79,41 @@
 (defn game-over? [board]
   (if (or (three-aligned? board) (board-full? board)) true false))
 
-(defn next-player-turn [board user-symbol]
+(defprotocol Player
+  (get-symbol [this])
+  (get-move [this board]))
+
+(deftype HumanPlayer [symbol]
```

Comment:
Have you considered separating the players into different classes/files? That way you'd have each class/file being responsible for their own single logic. You could do the same for the board and for the UI as well, and keep here just the core logic for the game

Reply:
Yeah, it's in the backlog, but it's not yet a priority so I've left it aside - I appreciate it makes it harder to read the code, though!